src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -159,25 +158,40 @@ object BlockVTK {
     }
     offsetIterator(offsets, localOffsets.toSeq)
   }
+
+  /*
+   * FIXME: This is very inefficient and we probably want to use the right data
+   * structures to deal with duplicates for us.
+   */
+  private def distinctArrays(vertices: Seq[Array[Double]], previous: Seq[Array[Double]]=Nil): Seq[Array[Double]] = {
+    vertices match {
+      case Nil => previous
+      case vert+:verts =>
+        if (previous.exists(previousVertex => previousVertex sameElements vert)) {
+          distinctArrays(verts, previous)
+        } else {
+          distinctArrays(verts, vert+:previous)
+        }
+    }
+  }
 }
 
 /**
   * Simple data structure to contain data that represents a rock block that can be turned into vtk format
-  * @param Rock block 
   * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
   */
 case class BlockVTK(block: Block) {
   val orientedVertices = BlockVTK.orientVertices(block)
-  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
-  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
-  val vertices = tupleVertices flatMap { t =>
-    List(t._1, t._2, t._3)
+  val arrayVertices = BlockVTK.distinctArrays(orientedVertices.values.flatten.toSeq)
```

Comment:
Based on your FIXME comment on Line 163, what do you suggest we do to get round iterating through the entire list of vertices to find the distinct ones?
How bad would it be to call `toSeq` on each of the vertices so we can use `distinct`? From a quick test in the Scala interpreter, it looks like it converts them to `WrappedArray`s; however, I have no idea if this would re-introduce the boxing problem we were having and whether this is more inefficient than what is being done right now.